src/Data/TypeRep/Vector.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE ExplicitForAll      #-}
+{-# LANGUAGE KindSignatures      #-}
+{-# LANGUAGE PolyKinds           #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies        #-}
+
+
+module Data.TypeRep.Vector
+       ( TypeRepVector (..)
+       , empty
+       , insert
+       , lookup
+       , size
+       ) where
+
+import Prelude hiding (lookup)
+
+import Data.Proxy (Proxy (..))
+import Data.Typeable (TypeRep, Typeable, typeRep, typeRepFingerprint)
+import Data.Word (Word64)
+import GHC.Base (Any)
+import GHC.Fingerprint (Fingerprint (..))
+import Unsafe.Coerce (unsafeCoerce)
+
+import qualified Data.Vector as V
+import qualified Data.Vector.Unboxed as Unboxed
+
+type FingerprintPair = (Word64, Word64)
```

Comment:
Nice trick, so you reuse the instance for pairs. Could we try to use an `Unboxed.Vector Fingerprint` directly, to avoid converting to/from tuples? If this works, you can submit a PR to the `vector` lib